### PR TITLE
Remove `simulate_tree()` and `simulate_tree_from_pop()` and replace usage with `simulate_chains()`

### LINKED ISSUE #181: Remove `simulate_tree()` and `simulate_tree_from_pop()`
Now that `simulate_tree()` and `simulate_tree_from_pop()` have been combined, there is no need to keep these two functions. We will remove all occurrences of these two functions and replace them with equivalent `simulate_chains()` calls.

Closing this issue will remove the need to resolve #158.

### PR DESCRIPTION
**Pull request by jamesmbaazam**
This PR closes #181 by:

- Unexporting and removing all occurrences of `simulate_tree()` and `simulate_tree_from_pop()`,
- Replacing their use with `simulate_chains()` in tests, vignettes, and examples.

Additionally, where necessary, descriptions involving `simulate_tree()` and `simulate_tree_from_pop()` have been rephrased. Examples include the README (64015e1e20568ac852cd1f29a25b1b6721ef733a) and Getting Started vignettes (e38453a58306466c2576c47b6558610d4be67072).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,25 +58,22 @@ library("epichains")
...
+  seed the outbreak, a distribution of offspring per case, and a chain
+  statistic to track (size or length/duration). It optionally accepts
+  other population related inputs such as the population size (defaults
+  to Inf) and percentage of the population initially immune (defaults to
+  0). This function returns an object with columns that track
+  information on who infected whom, the generation of infection and, if
+  a generation time function is specified, the time of infection.
+
+- `simulate_summary()`: provides a performant version of
+  `simulate_chains()` that only tracks and return a vector of realized
```
Comment (jamesmbaazam) [inline README.md:74]:
```suggestion
  `simulate_chains()` that only tracks and returns a vector of realized
```

**Review by sbfnk — APPROVED** [review]
LGTM

**Comment by sbfnk** [comment]
Also closes #158

**Comment by jamesmbaazam** [comment]
Thanks. #158 will be better addressed by #182.

**Comment by sbfnk** [comment]
Ah sorry yes that makes sense

**Pull request merged**